storm-client/src/jvm/org/apache/storm/utils/CuratorUtils.java

```diff
@@ -84,6 +89,120 @@ protected static void setupBuilder(CuratorFrameworkFactory.Builder builder, fina
         if (auth != null && auth.scheme != null && auth.payload != null) {
             builder.authorization(auth.scheme, auth.payload);
         }
+        boolean sslEnabled = ObjectReader.getBoolean(conf.get(Config.ZK_SSL_ENABLE), false);
+        if (sslEnabled) {
+            TruststoreKeystore truststoreKeystore = new TruststoreKeystore(conf);
+            ZKClientConfig zkClientConfig = new ZKClientConfig();
+            try {
+                setSslConfiguration(zkClientConfig, truststoreKeystore);
+            } catch (ConfigurationException e) {
+                throw new RuntimeException(e);
+            }
+            builder.zkClientConfig(zkClientConfig);
+        }
+    }
+
+    /**
+     * Configure ZooKeeper Client with SSL/TLS connection.
+     * @param zkClientConfig ZooKeeper Client configuration
+     * @param truststoreKeystore The truststore and keystore configs
+     */
+    private static void setSslConfiguration(ZKClientConfig zkClientConfig,
+                                            TruststoreKeystore truststoreKeystore) throws ConfigurationException {
+        setSslConfiguration(zkClientConfig, new ClientX509Util(), truststoreKeystore);
+    }
+
+    private static void setSslConfiguration(ZKClientConfig zkClientConfig,
```

Comment:
What is the reason for having two methods here? Both are `private`, so supplying a custom `ClientX509Util` isn't possible anyway?